src/main/kotlin/com/sourcegraph/cody/history/HistoryService.kt

```diff
@@ -23,6 +23,17 @@ class HistoryService(private val project: Project) :
     synchronized(listeners) { listeners += listener }
   }
 
+  @Synchronized
+  fun updateChatMessages(
```

Comment:
is it ok we do writing operations in a synchronized method within EDT?

Reply:
I am getting `Access is allowed from Event Dispatch Thread (EDT) only` when i try to execute it on background thread...